src/runtime_src/core/edge/drm/zocl/zocl_drv.c

```diff
@@ -671,26 +700,12 @@ static int zocl_drm_platform_probe(struct platform_device *pdev)
 
 	/* Initialzie IOMMU */
 	if (iommu_present(&platform_bus_type)) {
-		struct iommu_domain_geometry *geometry;
-		u64 start, end;
-		int ret = 0;
-
-		zdev->domain = iommu_domain_alloc(&platform_bus_type);
-		if (!zdev->domain)
-			return -ENOMEM;
-
-		ret = iommu_attach_device(zdev->domain, &pdev->dev);
-		if (ret) {
-			DRM_INFO("IOMMU attach device failed. ret(%d)\n", ret);
-			iommu_domain_free(zdev->domain);
-		}
-
-		geometry = &zdev->domain->geometry;
-		start = geometry->aperture_start;
-		end = geometry->aperture_end;
-
-		DRM_INFO("IOMMU aperture initialized (%#llx-%#llx)\n",
-				start, end);
+		/*
+		 * Note: we ignore the return value of zocl_iommu_init().
+		 * In the case of failing to initialize iommu, zocl
+		 * driver will keep working with iommu disabled.
+		 */
+		(void) zocl_iommu_init(zdev, pdev);
```

Comment:
Should we at least warn if the IOMMU attach fails?

Reply:
Yes we do log warning message if zocl_iommu_init() if attache fails